Trappist/src/Promact.Trappist.Test/Questions/QuestionsRepositoryTest.cs

```diff
@@ -214,31 +214,80 @@ public QuestionsRepositoryTest(Bootstrap bootstrap) : base(bootstrap)
 
             //Adding code snippet question
             var codingQuestion = await CreateCodingQuestion();
+            codingQuestion.CodeSnippetQuestion.CodeSnippetQuestionTestCases.Add(new CodeSnippetQuestionTestCases()
+            {
+                TestCaseTitle = "Default Check",
+                TestCaseDescription = "This case is default case",
+                TestCaseMarks = 10.00,
+                TestCaseType = TestCaseType.Necessary,
+                TestCaseInput = "1+1",
+                TestCaseOutput = "2",
+            });
             await _questionRepository.AddCodeSnippetQuestionAsync(codingQuestion, applicationUser.Id);
 
             var question = await _trappistDbContext
                 .Question
                 .FirstOrDefaultAsync(x => x.QuestionDetail == codingQuestion.Question.QuestionDetail);
 
-            codingQuestion.CodeSnippetQuestion.CheckCodeComplexity = false;
-            codingQuestion.CodeSnippetQuestion.CheckTimeComplexity = false;
-            codingQuestion.Question.Id = question.Id;
-            codingQuestion.Question.QuestionDetail = "Updated question details";
-            codingQuestion.Question.DifficultyLevel = DifficultyLevel.Hard;
-            codingQuestion.CodeSnippetQuestion.LanguageList = new string[] { "C" };
-            await _questionRepository.UpdateCodeSnippetQuestionAsync(question.Id, codingQuestion, applicationUser.Id);
+            var updatedQuestion = await _questionRepository.GetQuestionByIdAsync(question.Id);
+
+            //Updating code snippet question 
+            updatedQuestion.CodeSnippetQuestion.CheckCodeComplexity = false;
+            updatedQuestion.CodeSnippetQuestion.CheckTimeComplexity = false;
+            updatedQuestion.CodeSnippetQuestion.CodeSnippetQuestionTestCases.Remove(codingQuestion.CodeSnippetQuestion.CodeSnippetQuestionTestCases.First());
+            updatedQuestion.CodeSnippetQuestion.CodeSnippetQuestionTestCases.Add(new CodeSnippetQuestionTestCases
+            {
+                TestCaseTitle = "New check",
+                TestCaseDescription = "This is a new case",
+                TestCaseMarks = 10.00,
+                TestCaseType = TestCaseType.Basic,
+                TestCaseInput = "1+1",
+                TestCaseOutput = "2",
+            });
+            updatedQuestion.Question.QuestionDetail = "Updated question details";
+            updatedQuestion.Question.DifficultyLevel = DifficultyLevel.Hard;
+            updatedQuestion.CodeSnippetQuestion.LanguageList = new string[] { "C" };
+            updatedQuestion.Question.Id = question.Id;
+            await _questionRepository.UpdateCodeSnippetQuestionAsync(updatedQuestion, applicationUser.Id);
 
-            var updatedQuestion = await _trappistDbContext
+            var questionAfterUpdate = await _trappistDbContext
```

Comment:
if you are fetching record based on id then it is convenient way to use **Find** method